src/OrchardCore.Modules/OrchardCore.Alias/Handlers/AliasPartHandler.cs

```diff
@@ -108,11 +110,16 @@ private async Task<string> GenerateUniqueAliasAsync(string alias, AliasPart cont
             while (true)
             {
                 var versionedAlias = $"{unversionedAlias}-{version++}";
-                if ((await _session.QueryIndex<AliasPartIndex>(o => o.Alias == versionedAlias && o.ContentItemId != context.ContentItem.ContentItemId).CountAsync()) == 0)
+                if (await IsAliasUniqueAsync(versionedAlias, context))
                 {
                     return versionedAlias;
                 }
             }
         }
+
+        private async Task<bool> IsAliasUniqueAsync(string alias, AliasPart context)
```

Comment:
We can use inline lambda expression instead

Reply:
check the code, the method is used twice

Reply:
Even though

```csharp
private async Task<bool> IsAliasUniqueAsync(string alias, AliasPart context)
    => (await _session.QueryIndex<AliasPartIndex>(o => o.Alias == alias && o.ContentItemId != context.ContentItem.ContentItemId).CountAsync()) == 0;
```

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/local-functions-vs-lambdas

I'm happy with it as a method.

Reply:
but it's not a local method :smile:

Reply:
It's an expression bodied function that results to the same IL code ;)

It allows more compact code, but not sure it is always more readable and sometimes i don't use it when there is a good chance that i have to add a line of code and then to re-write the full body.

But yes in some contexts i like to use them. If it becomes for us a mandatory syntax rule, maybe could be part of a refactoring work, they are many places that could be updated.

But here i don't think it should be a blocking point.